cylc/flow/scheduler.py

```diff
@@ -1896,3 +1900,27 @@ async def handle_exception(self, exc: Exception) -> NoReturn:
         """
         await self.shutdown(exc)
         raise exc from None
+
+    def validate_finalcp(self):
+        """Warn if Stop Cycle point is on or after the final cycle point
+        """
+        if not self.options.fcp:
```

Comment:
`self.config.final_point` is used elsewhere in `process_cylc_stop_point()`
```suggestion
        if self.config.final_point is None:
```